tf-ncl/src/nickel_builder.rs

```diff
@@ -17,77 +19,94 @@ pub struct Complete(Option<RichTerm>);
 pub struct Field<RB> {
     record: RB,
     path: StaticPath,
-    metadata: Option<MetaValue>,
+    metadata: FieldMetadata,
+}
+
+pub struct Types(pub TypeF<Box<Types>, RecordRows, EnumRows>);
+
+impl From<Types> for TypeF<Box<Types>, RecordRows, EnumRows> {
```

Comment:
I wonder if we should implement a generic version of this directly in Nickel. That is:

```rust
impl<RRows, ERows> From<Types> for TypeF<Box<Types>, RRows, ERows> {
 // ...
}
```

Reply:
I was using this in order to make the builder interface slightly more ergonomic. First note, that I'm using a custom `Types` for tying the knot which doesn't carry position information. In the field builder I have a function
```rust
pub fn contract(mut self, contract: impl Into<TypeF<Box<Types>, RecordRows, EnumRows>>) -> Self {
    self.metadata.annotation.contracts.push(LabeledType {
        types: add_positions(Types(contract.into())),
        label: Default::default(),
    });
    self
}
```
so that I can just write `field.contract(TypeF::Num)` for instance. But in fact, it feels more logical to me now to have `contract` take an `impl Into<Types>` and turn the `From` instance around.

Anyway, I don't think implementing this generically in Nickel makes much sense, does it? Do we have a lot of conversions of this kind happening?